Packages/com.unity.inputsystem/InputSystem/Devices/NoiseFilter.cs

```diff
@@ -0,0 +1,246 @@
+                    elementsToAdd[elementCount++] = newElement;
+                }
+                else
+                {
+                    InputControl<float> controlAsFloat = control as InputControl<float>;
+                    if (controlAsFloat != null && controlAsFloat.processors != null)
+                    {
+                        if (controlAsFloat.processors != null)
+                        {
+                            FilteredElement newElement;
+                            newElement.controlIndex = i;
+                            newElement.type = EElementType.TypeFloat;
+                            InputStateBlock stateblock = control.stateBlock;
+                            elementsToAdd[elementCount++] = newElement;
+                        }
+                    }
+                    else
+                    {
+                        InputControl<Vector2> controlAsVec2 = control as InputControl<Vector2>;
+                        if (controlAsVec2 != null && controlAsVec2.processors != null)
+                        {
+                            FilteredElement newElement;
+                            newElement.controlIndex = i;
+                            newElement.type = EElementType.TypeVec2;
+                            InputStateBlock stateblock = control.stateBlock;
+                            elementsToAdd[elementCount++] = newElement;
+                        }
+                    }
+                }
+            }
+
+            filter.elements = new FilteredElement[elementCount];
+            for (int j = 0; j < elementCount; j++)
+            {
+                filter.elements[j] = elementsToAdd[j];
+            }
+
+            return filter;
+        }
+
+        /// <summary>
+        /// The list of elements to be checked for.  Each element represents a single InputControl.
+        /// </summary>
+        public FilteredElement[] elements;
+
+        /// <summary>
+        /// Called when the NoiseFilter gets applied to a device, calls down to any individual FilteredElements that need to do any work.
+        /// </summary>
+        /// <param name="device">The device you want to apply filtering to.</param>
+        internal void Apply(InputDevice device)
+        {
+            if (device == null)
+                throw new ArgumentException("No device supplied to apply NoiseFilter to.", "device");
+
+            IntPtr noiseFilterPtr = InputStateBuffers.s_NoiseFilterBuffer;
+            if (noiseFilterPtr != IntPtr.Zero)
+            {
+                BitmaskHelpers.Whitelist(noiseFilterPtr, device);
+
+                for (int i = 0; i < elements.Length; i++)
+                {
+                    elements[i].Apply(noiseFilterPtr, device);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resets a device to unfiltered
+        /// </summary>
+        /// <param name="device">The device you want reset</param>
+        internal void Reset(InputDevice device)
```

Comment:
Again, I'd like to keep it named more on the context it's called in, so that future edits don't need function naming changes.  The NoiseFilter Resets here, what it resets is private knowledge.